p2p/src/main/java/bisq/network/p2p/peers/PeerManager.java

```diff
@@ -664,11 +664,11 @@ public void shutDownConnection(NodeAddress peersNodeAddress, CloseConnectionReas
                     // If we have a new connection the supportedCapabilities is empty.
                     // We lookup if we have already stored the supportedCapabilities at the persisted or reported peers
                     // and if so we use that.
-                    if (!supportedCapabilities.hasCapabilities()) {
+                    if (supportedCapabilities.isEmpty()) {
                         Set<Peer> allPeers = new HashSet<>(getPersistedPeers());
                         allPeers.addAll(getReportedPeers());
                         Optional<Peer> ourPeer = allPeers.stream().filter(peer -> peer.getNodeAddress().equals(connection.getPeersNodeAddressOptional().get()))
-                                .filter(peer -> !peer.getCapabilities().hasCapabilities())
+                                .filter(peer -> !peer.getCapabilities().isEmpty())
```

Comment:
Here is the boolean logig inverse to the older version.

Reply:
This, again, is a bug which has been introduced with the previous refactoring.

The goal is to find "cached" capabilities for a given peer. By filtering for no capabilities (`!peer...hasCapabilities`), we only get a list of peers which have an empty set of capabilities - which does not help in achieving the goal. Again, this bug does not break anything as active peers (with time) report their capabilities and hence, the bug does not show itself easily.

This change, again, restores the intended functionality by filtering for peers with a non-empty (`!peer...isEmpty`) set of capabilities.